src/coreclr/jit/gentree.cpp

```diff
@@ -18163,6 +18163,10 @@ CORINFO_CLASS_HANDLE Compiler::gtGetClassHandle(GenTree* tree, bool* pIsExact, b
             *pIsExact = true;
             objClass  = exactClass;
         }
+        else
+        {
+            *pIsExact = impIsClassExact(objClass);
+        }
```

Comment:
I think this change is enough, isn't? Can you check whether diffs are the same if you remove changes inside SetClass/UpdateClass ?

Reply:
I'd prefer to avoid doing that, updating the info on the localvar side leads to better local typing in other usages, is clearly shown in jitdisasm on the locals table and avoids having to check this on every `gtGetClassHandle` call. I've added this just for the other sources of types here.

Reply:
if they don't add value then it means on every call to SetClass/UpdateClass you added redundant VM call (we don't have a good metric to estimate the real JIT TP with them currently)